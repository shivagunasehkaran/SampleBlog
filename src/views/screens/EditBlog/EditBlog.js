import React, {Component} from 'react';
import {View, Text, TextInput, TouchableOpacity, StyleSheet} from 'react-native';
import theme from '../../../themes';

class EditBlog extends Component {
    static navigationOptions = ({navigation}) => ({
        title: ('Edit blogs'),
        headerStyle: {
            backgroundColor: theme.colors.buttonColor,
        },
    });

    constructor(props) {
        super(props);
        this.state = {
            title: '',
            content: '',
        };
    }

    handleTitle = (text) => {
        this.setState({title: text});
    };

    handleContent = (text) => {
        this.setState({content: text});
    };

    save = (title, content) => {
        alert('title: ' + title + ' content: ' + content);
    };

    renderBlogEditForm = () => {
        return (
            <View style={styles.container}>
                <Text style={styles.title}>
                    {('Enter New Title')}
                </Text>
                <TextInput style={styles.input}
                           underlineColorAndroid="transparent"
                           placeholder="Title"
                           placeholderTextColor="#686868"
                           autoCapitalize="none"
                           onChangeText={this.handleTitle} />
                <Text style={styles.title}>
                    {('Enter New Content')}
                </Text>
                <TextInput style={styles.input}
                           underlineColorAndroid="transparent"
                           placeholder="Content"
                           placeholderTextColor="#686868"
                           autoCapitalize="none"
                           onChangeText={this.handleContent} />

                <TouchableOpacity
                    style={styles.saveButton}
                    onPress={
                        () => this.save(this.state.title, this.state.content)
                    }>
                    <Text style={styles.saveButtonText}> {('Save')} </Text>
                </TouchableOpacity>
            </View>
        );
    };

    render() {
        return (
            <View style={{flex: 1}}>
                {this.renderBlogEditForm()}
            </View>
        );
    }
}

const styles = StyleSheet.create({
    container: {
        paddingTop: 10,
    },
    title: {
        fontSize: 15,
        paddingLeft: 15,
        color: theme.colors.black,
    },
    input: {
        margin: 15,
        height: 40,
        borderColor: theme.colors.black,
        borderWidth: 1,
    },
    saveButton: {
        backgroundColor: theme.colors.buttonColor,
        padding: 10,
        margin: 15,
        height: 40,
    },
    saveButtonText: {
        fontSize: 15,
        fontWeight: '500',
        color: theme.colors.black,
    },
});

export default EditBlog;
