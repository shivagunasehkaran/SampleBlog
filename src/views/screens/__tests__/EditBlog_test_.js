import React from 'react';
import renderer from 'react-test-renderer';
import EditBlog from '../EditBlog/EditBlog';

test('renders correctly edit blog', () => {
    const tree = renderer.create(<EditBlog />).toJSON();
    expect(tree).toMatchSnapshot();
});
