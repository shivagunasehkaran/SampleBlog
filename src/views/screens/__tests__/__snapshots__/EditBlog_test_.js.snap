// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "paddingTop": 10,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "black",
          "fontSize": 15,
          "paddingLeft": 15,
        }
      }
    >
      Enter New Title
    </Text>
    <TextInput
      allowFontScaling={true}
      autoCapitalize="none"
      onChangeText={[Function]}
      placeholder="Title"
      placeholderTextColor="#686868"
      rejectResponderTermination={true}
      style={
        Object {
          "borderColor": "black",
          "borderWidth": 1,
          "height": 40,
          "margin": 15,
        }
      }
      underlineColorAndroid="transparent"
    />
    <Text
      style={
        Object {
          "color": "black",
          "fontSize": 15,
          "paddingLeft": 15,
        }
      }
    >
      Enter New Content
    </Text>
    <TextInput
      allowFontScaling={true}
      autoCapitalize="none"
      onChangeText={[Function]}
      placeholder="Content"
      placeholderTextColor="#686868"
      rejectResponderTermination={true}
      style={
        Object {
          "borderColor": "black",
          "borderWidth": 1,
          "height": 40,
          "margin": 15,
        }
      }
      underlineColorAndroid="transparent"
    />
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "orange",
          "height": 40,
          "margin": 15,
          "opacity": 1,
          "padding": 10,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "black",
            "fontSize": 15,
            "fontWeight": "500",
          }
        }
      >
         
        Save
         
      </Text>
    </View>
  </View>
</View>
`;

exports[`renders correctly edit blog 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "paddingTop": 10,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "black",
          "fontSize": 15,
          "paddingLeft": 15,
        }
      }
    >
      Enter New Title
    </Text>
    <TextInput
      allowFontScaling={true}
      autoCapitalize="none"
      onChangeText={[Function]}
      placeholder="Title"
      placeholderTextColor="#686868"
      rejectResponderTermination={true}
      style={
        Object {
          "borderColor": "black",
          "borderWidth": 1,
          "height": 40,
          "margin": 15,
        }
      }
      underlineColorAndroid="transparent"
    />
    <Text
      style={
        Object {
          "color": "black",
          "fontSize": 15,
          "paddingLeft": 15,
        }
      }
    >
      Enter New Content
    </Text>
    <TextInput
      allowFontScaling={true}
      autoCapitalize="none"
      onChangeText={[Function]}
      placeholder="Content"
      placeholderTextColor="#686868"
      rejectResponderTermination={true}
      style={
        Object {
          "borderColor": "black",
          "borderWidth": 1,
          "height": 40,
          "margin": 15,
        }
      }
      underlineColorAndroid="transparent"
    />
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "orange",
          "height": 40,
          "margin": 15,
          "opacity": 1,
          "padding": 10,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "black",
            "fontSize": 15,
            "fontWeight": "500",
          }
        }
      >
         
        Save
         
      </Text>
    </View>
  </View>
</View>
`;
